### ISSUE TITLE
allow configuration of character count in string diff formatting

### ISSUE THREAD

**Issue by mitchdraft** [onsi/gomega#346]
from `format/format.go`:
```go

/*

Generates a nicely formatted matcher success / failure message

Much like Message(...), but it attempts to pretty print diffs in strings

Expected
    <string>: "...aaaaabaaaaa..."
to equal               |
    <string>: "...aaaaazaaaaa..."

*/

func MessageWithDiff(actual, message, expected string) string {
```

and https://github.com/onsi/gomega/blob/master/format/format.go#L146

we get 5 characters of context around the first diff in a string comparison assertion error message.

5 characters is often enough, however when comparing large structs, as can be the case with protobuf messages [example](https://github.com/solo-io/build/pull/6/files#diff-abb9a51fd474a20745b8abe8d467cbc3R23) it would be useful to have more context. Otherwise, long values can obscure the key, and make it harder to back-out the issue.

**Suggestion**: export `format.CharactersAroundMismatchToInclude`

**Comment by williammartin** [comment]
Thanks for the issue. Just to check, the `format.TruncatedDiff` value set to false isn't satisfactory for you here?

**Comment by mitchdraft** [comment]
Thanks for pointing that out, I hadn't considered it.
Still, I find the truncation super helpful on large structs. It has offered a nice balance of ease-of-test creation and usefulness of error messages.

In this [util](https://github.com/solo-io/build/blob/master/test/testutils/proto_compare.go#L23), for example. [here are my notes on the alternatives](https://github.com/solo-io/build/commit/469800487c0bca6adda8bd5f3f952df2a983fad3#diff-a86882f3ee3b354a5f04fa5e763a1c11R65) - feedback welcome, maybe there's a better way

**Comment by mitchdraft** [comment]
additional note - we also decided to avoid `MatchFields` so as to avoid overlooking any new fields that might be added in the future

**Comment by blgm** [comment]
This was resolved by #347
